bin/git-sync

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+function _usage()
+{
+  local command="git sync"
+  cat << EOS
+Usage:
+  ${command} [<remote> <branch>]
+  ${command} -h | --help
+
+Sync local branch with <remote>/<branch>.
+When <remote> and <branch> are not specified on the command line, upstream of local branch will be used by default.
+All changes and untracked files and directories will be removed.
+
+Examples:
+  Sync with upstream of local branch:
+    ${command}
+
+  Sync with origin/master:
+    ${command} origin master
+EOS
+}
+
+function main()
+{
+  while [ "$1" != "" ]; do
+    case $1 in
+      -h | --help)
+        _usage
+        exit
+        ;;
+      * )
+        if [ "${remote}" = "" ]; then
+          local remote="$1"
+        elif [ "${branch}" = "" ]; then
+          local branch="$1"
+        else
+          echo -e "Error: too many arguments.\n"
+          _usage
+          exit 1
+        fi
+        ;;
+    esac
+    shift
+  done
+
+  if [ "${remote}" = "" ]; then
+    local remote_branch="$(git rev-parse --abbrev-ref --symbolic-full-name @{u})"
+    branch="$(git rev-parse --abbrev-ref --symbolic-full-name @)"
+    remote="${remote_branch%%/${branch}}"
```

Comment:
A little quibble:
`git rev-parse --abbrev-ref --symbolic-full-name @{u}` will return nothing if there is not related upstream branch. 
Then users will see `... & sync with ''? [y/N]` and the command may simply reset to current branch, which is not really wanted.